File: agents/typescript/example-workflows/src/create-workflow.ts
import {workflows} from "./workflows";
import * as readline from 'readline';
import {AsteroidAgents} from "asteroid-odyssey";
import {CreateWorkflowRequest} from "asteroid-odyssey/dist/generated/agents";

const apiKey: string = process.env.ASTEROID_API_KEY ?? "";
const asteroid = new AsteroidAgents(apiKey);

/**
 * Ceres is our browser running agent that can execute tasks for you. This is the only one at the moment,
 * but there's more coming ASAP.
 */
const agentName: string = "ceres";

async function createWorkflow(request: CreateWorkflowRequest) {
  try {
    const workflowId = await asteroid.createWorkflow(agentName, request);
    console.log('Workflow created with ID:', workflowId);
  } catch (error) {
    console.error('Error executing the workflow:', error);
  }
}

function chooseWorkflow(): Promise<typeof workflows[0]> {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  workflows.forEach((workflow, index) => {
    console.log(`${index + 1}. ${workflow.name}`);
  });

  return new Promise((resolve) => {
    rl.question('Choose one of the above workflows: ', (choiceNumber) => {
      const choice = parseInt(choiceNumber);
      rl.close(); // Close the readline interface
      resolve(workflows[choice - 1]);
    });
  });
}

chooseWorkflow().then((workflow) => {
  createWorkflow(workflow);
});

File: agents/python/example-workflows/create_workflow.py
import os
from asteroid_odyssey import Odyssey
from asteroid_odyssey.api.generated.agents.asteroid_agents_api_client.models import CreateWorkflowRequest
from workflows import workflows

api_key = os.getenv('ASTEROID_API_KEY', '')
asteroid = Odyssey(api_key)

agent_name = "ceres"

def create_workflow(request: CreateWorkflowRequest):
    try:
        workflow_id = asteroid.create_workflow(agent_name, request)
        print('Workflow created with ID:', workflow_id)
    except Exception as error:
        print('Error executing the workflow:', error)

def choose_workflow():
    for index, workflow in enumerate(workflows):
        print(f"{index + 1}. {workflow.name}")

    choice_number = input('Choose one of the above workflows: ')
    choice = int(choice_number)
    return workflows[choice - 1]

if __name__ == '__main__':
    workflow = choose_workflow()
    create_workflow(workflow)
